src/components/narrativeEditor/markdownInput.js

```diff
@@ -0,0 +1,140 @@
+        const msg = `Error obtaining narratives from ${narrativeFile.name}. Details: ${err.message}`;
+        console.error(msg);
+        setError(msg);
+        return;
+      }
+      updateApiCalls(narrativeData.datasets);
+      setNarrative(narrativeData);
+    }
+    function handleDragover(event) {
+      event.preventDefault();
+    }
+    function handleDragEnter() {
+      setIsDragging(true);
+    }
+    function handleDragLeave() {
+      setIsDragging(false);
+    }
+    const domElement = el.current; // important for the clean up function
+    domElement.addEventListener("dragover", handleDragover, false);
+    domElement.addEventListener("drop", handleDrop, false);
+    domElement.addEventListener("dragenter", handleDragEnter, false);
+    domElement.addEventListener("dragleave", handleDragLeave, false);
+
+    return () => {
+      domElement.removeEventListener("dragover", handleDragover, false);
+      domElement.removeEventListener("drop", handleDrop, false);
+      domElement.removeEventListener("dragenter", handleDragEnter, false);
+      domElement.removeEventListener("dragleave", handleDragLeave, false);
+
+    };
+  }, [setNarrative, setError, isDragging, setIsDragging]);
+
+  return (
+    <Styles.FilePickerContainer ref={el}>
+      <Styles.FilePickerHeader>
+        <h4>Provide a narrative to analyse</h4>
+      </Styles.FilePickerHeader>
+      <Styles.FilePickerBody dragActive={isDragging}>
+        <Styles.FileIcon size={70}/> {/* todo: dragActive prop */}
+        {fileName ? (
+          <Styles.DraggedText>
+            <p>{`Currently analysing ${fileName}`}</p>
+            <p>Drag & Drop another markdown file to restart</p>
+          </Styles.DraggedText>
+        ) : (
+          <Styles.DraggedText>
+            Drag & Drop markdown file to begin
+          </Styles.DraggedText>
+        )}
+      </Styles.FilePickerBody>
+    </Styles.FilePickerContainer>
+  );
+
+};
+
+/**
+ * In nextstrain.org & auspice we use local pathnames, e.g. /zika
+ * But here I deliberately want to be able to access datasets on nextstrain.org
+ * (remember we're developing this from within auspice, which can't access /community URLs etc)
+ * Since we are running this on localhost we can't actually hit nextstrain.org
+ * endpoints due to CORS. Fair enough. So we rely on a local (development)
+ * nextstrain.org server running at localhost:5000 AND running this auspice instance
+ * on port 8000 as that combination has cors disabled. We can also request
+ * them from the current (auspice) server which is useful for development to avoid
+ * too many requests across the wire, but note that it'll only work for dataset
+ * paths you have on disk.
+ */
+function updateApiCalls(datasets) {
+  for (const datasetName of Object.keys(datasets)) {
+    const apiCalls = datasets[datasetName].apiCalls;
+    for (const [fileType, previousURL] of Object.entries(apiCalls)) {
+      apiCalls[fileType] = `${SERVER_TO_USE}${previousURL}`;
```

Comment:
_non-blocking_

This will break if `previousURL` _isn't_ just a path name but is a complete URL, which I believe is possible when the `serverAddress` extension is used. Consider using `URL()` instead, e.g. to only modify calls that have relative paths:

```suggestion
      if (SERVER_TO_USE) {
        apiCalls[fileType] = (new URL(previousURL, SERVER_TO_USE)).toString();
      }
```

or to modify all calls:

```suggestion
      if (SERVER_TO_USE) {
        const {pathname, search} = new URL(previousURL, "https://example.com");
        const newURL = new URL(SERVER_TO_USE);
        newURL.pathname = pathname;
        newURL.search = search;
        apiCalls[fileType] = newURL.toString();
      }
```

(both untested, but something like that)

Reply:
Yeah - this worked as-is so I just switched to it (https://github.com/nextstrain/auspice/compare/3ddbd1616ce697044c3c690eebc054226681335f..1adfb5c737d3393ddf22081c94cda3de8b4f7913). I also split it out into a separate file and commented it to indicate that this is only ever intended to development purposes.